File: python/2.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# @Author: ben_cripps
# @Date:   2014-12-06 14:47:20
# @Last Modified by:   ben_cripps
# @Last Modified time: 2015-01-02 22:44:34

def fib(limit):
    arr = []

    def loop(a,b):
        if (a < limit):
            b+= a
            arr.append(b)
            loop(b,a)

    loop(0,1)
    return arr

     

print(reduce(lambda x,y: x+y, filter(lambda x: x % 2 == 0, fib(4000000))))

File: c#/2.cs
using System;

public class Euler
{   
    private int Fib(int limit, int sum = 0, int a = 0, int b = 1) 
    {   
        int temp = 0;

        if (a < limit)
        {
            if (a % 2 == 0) 
            {
                sum+= a;
            }
            
            temp = a;
            a = b;
            b = temp + b;
            return Fib(limit, sum, a, b);
        }
        
        return sum;
    }
    
    public void Main()
    {
        Console.WriteLine(Fib(4000000));
    }
}